broker/lib/controllers/ServiceFabrikApiController.js

```diff
@@ -249,45 +343,224 @@ class ServiceFabrikApiController extends FabrikBaseController {
+                  operation: CONST.OPERATION_TYPE.BACKUP
+                }
+              })
+              .then(() => {
+                lockedDeployment = true;
+                return eventmesh.server.createOperationResource({
+                  resourceId: req.params.instance_id,
+                  operationName: CONST.OPERATION_TYPE.BACKUP,
+                  operationType: CONST.APISERVER.RESOURCE_NAMES.DEFAULT_BACKUP,
+                  operationId: backupGuid,
+                  value: backupOptions
+                });
+              });
+          });
+      })
+      .then(() => {
+        backup_started_at = new Date();
+        //check if resource exist, else create and then update
+        return Promise.try(() => eventmesh.server.getResource('deployment', 'directors', req.params.instance_id))
+          .catch(() => eventmesh.server.createDeploymentResource(null, req.params.instance_id, {}))
+          .then(() => eventmesh.server.updateLastOperation({
+            resourceId: req.params.instance_id,
+            operationName: CONST.OPERATION_TYPE.BACKUP,
+            operationType: CONST.APISERVER.RESOURCE_NAMES.DEFAULT_BACKUP,
+            value: backupGuid
+          }))
+          .then(() => ServiceFabrikApiController.getResourceOperationStatus({
+            operationId: backupGuid,
+            start_state: CONST.APISERVER.RESOURCE_STATE.IN_QUEUE,
+            started_at: backup_started_at
+          }));
+      })
+      .tap(status => {
+        logger.info(`Backup Response `, status.response);
+        return req.manager
+          .findNetworkSegmentIndex(req.params.instance_id)
+          .then(networkIndex => {
+            logger.error('NetworkIndex is ', req.params, networkIndex);
+            return req.manager.getDeploymentName(req.params.instance_id, networkIndex);
+          });
+      })
+      .then(status => {
+        logger.info('Operation response:', status.response);
+        const body = JSON.parse(status.response);
+        res.status(CONST.HTTP_STATUS_CODE.ACCEPTED).send(body);
+      })
+      .catch(err => {
+        logger.info('Handling error :', err);
+        if (err instanceof EtcdLockError) {
+          throw err;
+        }
+        if (lockedDeployment) {
+          return lockManager.unlock(req.params.instance_id)
+            .throw(err);
+        }
+        throw err;
+      })
+      .catch(Timeout, () => {
+        return this.abortLastBackup(req, res);
+      });
+  }
+
   getLastBackup(req, res) {
+    if (config.enable_service_fabrik_v2) {
+      return this.getLastBackup20(req, res);
+    }
+    return this.getLastBackup10(req, res);
+  }
+
+  getLastBackup20(req, res) {
+    return eventmesh.server.getLastOperation({
+        resourceId: req.params.instance_id,
+        operationName: CONST.OPERATION_TYPE.BACKUP,
+        operationType: CONST.APISERVER.RESOURCE_NAMES.DEFAULT_BACKUP
+      })
+      .then(backup_guid =>
+        eventmesh.server.getOperationResult({
```

Comment:
Can the getOperationResult not return a parsed object rather than requiring clients to invoke `JSON.parse`?

Reply:
Some operationtype may choose to store just a string as result